src/c/service.c

```diff
@@ -856,9 +855,34 @@ void devsdk_service_start (devsdk_service_t *svc, iot_data_t *driverdfls, devsdk
     iot_log_error (svc->logger, "Unable to load config file: %s", err->reason);
     return;
   }
-  configmap = edgex_config_defaults (driverdfls, svc->name);
-  edgex_device_overrideConfig_map (configmap, config_file);
-  edgex_device_overrideConfig_env (svc->logger, configmap);
+  common_config_map = edgex_common_config_defaults (svc->name);
+  private_config_map = edgex_private_config_defaults(driverdfls);
+  edgex_device_overrideConfig_map (common_config_map, config_file);
+  edgex_device_overrideConfig_map (private_config_map, config_file);
+  edgex_device_overrideConfig_env (svc->logger, common_config_map);
+  edgex_device_overrideConfig_env (svc->logger, private_config_map);
+
+  configmap = iot_data_alloc_map (IOT_DATA_STRING);
+  iot_data_map_merge(configmap, common_config_map);
+  iot_data_map_merge(configmap, private_config_map);
+  iot_data_free(common_config_map);
+
+  /* Set up SecretStore */
+
+  char *secure = getenv (SECUREENV);
+  if (secure && strcmp (secure, "false") == 0)
```

Comment:
Add `bool secureMode;` to the service struct and set it here, then we can remove the other instances of getenv/strcmp